### PR TITLE
ParrotMock implemented with tests

### PR DESCRIPTION
**Pull request by Farlimb**
Adds both the Parrot entity

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.
- [X] Unit tests added (if applicable).

### CODE REVIEW

File: src/test/java/be/seeseemelk/mockbukkit/entity/ParrotMockTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		parrot.setSitting(true);
+		assertTrue(parrot.isSitting());
+	}
+	@Test
+	public void testSetOwner()
+	{
+		AnimalTamer owner = server.addPlayer();
+		parrot.setOwner(owner);
+		assertEquals(owner, parrot.getOwner());
+	}
```
Comment (Nirbose) [inline src/test/java/be/seeseemelk/mockbukkit/entity/ParrotMockTest.java]: remove all `public` to stay in the same style as the other unit test files and your `testGetType()`.

File: src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	public void setVariant(@NotNull Variant variant)
+	{
+		Preconditions.checkNotNull(variant, "Variant cannot be null");
+		this.variant = variant;
+	}
+
+	@Override
+	public boolean isDancing()
+	{
+		return this.isDancing;
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java]: This is always going to return false. You should probably add an access method for the isDancing field.

Comment (Farlimb) [reply src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java]: you mean a setter method? or a toogle method would be good one?

Comment (thelooter) [reply src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java]: Add a `setDancing(boolean)` method, that sets the `isDancing`field

File: src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	public void setDancing(boolean isDancing)
+	{
+		this.isDancing = isDancing;
+	}
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java]: `#isDancing` only returns true when there's a jukebox playing a record nearby. I don't believe we have jukeboxes implemented yet, so this shouldn't be implemented until we do.

**Review by Insprill — CHANGES_REQUESTED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java (hunk 3 above)
Comment (Nirbose) [reply src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java]: The juckbox is well implemented, but there are 3 methods that aren't "done", including `startPlaying`, and I wonder if it isn't this method that sets the peroquer's dancing to `true` (I haven't checked on papermc yet).

Comment (Farlimb) [reply src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java]: Should I add TO DO comment in setDancing method and make it throw new UnimplementedOperationException()?

Comment (Insprill) [reply src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java]: setDancing should be removed, and isDancing should throw, along with a comment about why it isn't implemented.

**Comment by Nirbose** [comment]
it would be nice to respect the code style of commit 79914cdf6bac6383ed2d521e7e8e21cec96bbd7d in the `ParotMockTest` file no?

File: src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+	@Override
+	public void setTamed(boolean tame)
+	{
+		this.isTamed = tame;
+	}
+
+	@Override
+	public @Nullable UUID getOwnerUniqueId()
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java:67]: Please add a Test for this Method

**Review by thelooter — CHANGES_REQUESTED** [review]

**Review by thelooter — APPROVED** [review]
LGTM

**Review by Farlimb — COMMENTED** [review]
done

**Review by Insprill — APPROVED** [review]
LGTM; Thank you!

**Pull request merged**